### ISSUE TITLE
Stub file issues when importing `ldpc` from another project

### ISSUE THREAD

**Issue by perlinm** [quantumgizmos/ldpc_v2#7]
I have another project that imports `ldpc`.  If I run `mypy` on that project, I get an error like
```
.../ldpc_v2/src_python/ldpc/bp_decoder/__init__.pyi:7: error: expected an indented block
after function definition on line 6  [syntax]
    class BpDecoderBase:
     ^
Found 1 error in 1 file (errors prevented further checking)
```
If I examine the offending file, lines 6 and 7 are
```
def io_test(pcm: Union[scipy.sparse.spmatrix,np.ndarray]):
class BpDecoderBase:
```
which is clearly off.  I can fix this (and a few more errors like it in the file) by changing it to something like
```
def io_test(pcm: Union[scipy.sparse.spmatrix,np.ndarray]): ...
class BpDecoderBase:
```
(that is, adding the ellipses `...` or `pass` to the end of `io_test`, but I eventually run into lines 321-322:
```
    def __init__(self, pcm: Union[np.ndarray, scipy.sparse.spmatrix], error_rate: Optional[float] = None,
    def decode(self, input_vector: np.ndarray) -> np.ndarray:
```
Here line `321` is clearly invalid.  I can try simply appending `): ...` to that line, but then I get more similar issues in other files.

Are these stubs generated automatically?  Is it at all easy to fix them?

**Comment by perlinm** [comment]
While I'm at it, I'll mention here that I also get a `DeprecationWarning` from imports of `ldpc`:
```
../ldpc_v2/src_python/ldpc/__init__.py:1: 16 warnings
  [...]/ldpc_v2/src_python/ldpc/__init__.py:1: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    import pkg_resources
```
Update: this is fixed by #8

**Comment by quantumgizmos** [comment]
Hi Michael. Stub files are a nightmare when using Cython. They're not generated automatically. The solution I found was to generate them manually with a script in the setup.py file https://github.com/quantumgizmos/ldpc_v2/blob/eca2e87b2cff1e40e5bf7af82334f82f5694f916/setup.py#L11

Could you implement the fix there?

Also, we have moved development of the LDPCv2 package to the main repository:

https://github.com/quantumgizmos/ldpc

This means that the version that is now installed from pip is now LDPCv2. So could you push fixes there? https://github.com/quantumgizmos/ldpc
